File: webManageProject/src/api/participant.ts
import request from '@/utils/request'

export interface Participant {
  id: number
  userId: number
  sportsMeetingId: number
  teamId: number | null
  teamName: string | null
  userCode: string
  name: string
  phone: string
  gender: string
  college: string
  major: string  // 专业/单位
  perPersonLimit?: number
}

export function getParticipantList(sportsMeetingId: number) {
  return request.get('/admin/participant/list', { params: { sportsMeetingId } })
}

export function getParticipantListByTeam(teamId: number) {
  return request.get('/admin/participant/listByTeam', { params: { teamId } })
}

export function addParticipant(data: Partial<Participant>) {
  return request.post('/admin/participant/add', data)
}

export function updateParticipant(data: Partial<Participant>) {
  return request.put('/admin/participant/update', data)
}

export function deleteParticipant(id: number) {
  return request.delete(`/admin/participant/${id}`)
}

export function removeFromTeam(participantId: number) {
  return request.put('/admin/participant/clearTeam', null, { params: { participantId } })
}

export function importParticipants(file: File, sportsMeetingId: number) {
  const formData = new FormData()
  formData.append('file', file)
  formData.append('sportsMeetingId', String(sportsMeetingId))
  return request.post('/admin/participant/import', formData, {
    headers: { 'Content-Type': 'multipart/form-data' }
  })
}

export function downloadImportTemplate() {
  return request.get('/admin/participant/import-template', { responseType: 'blob' })
}

File: SportBackend/src/main/java/com/dlust/sportbackend/entity/Participant.java
package com.dlust.sportbackend.entity;

import lombok.Data;
import java.time.LocalDateTime;

@Data
public class Participant {
    private Long id;
    private Long sportsMeetingId;
    private Long userId;       // 关联 user.id（替代原 userCode/name/...）
    private Long teamId;       // 代表队（后续分配）
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
    // 以下为 JOIN 查询时的扩展字段（非 participant 表字段）
    private String userCode;   // JOIN user
    private String name;       // JOIN user
    private String gender;     // JOIN user
    private String phone;      // JOIN user
    private String college;    // JOIN user
    private String major;      // JOIN user
    private String teamName;   // JOIN team
    private String sportsMeetingName;  // JOIN sports_meeting
}
